### ISSUE TITLE
support nested keys via the helper

### ISSUE THREAD

**Issue by mindw** [repeatedly/fluent-plugin-record-modifier#44]
A revival of #15 .
The record_accessor helper plugin is available for some time, perhaps this can be implemented 
cleanly now?

Thanks!

```
#<filter **>
#  @type record_modifier
#  <record>
#    log ${record['log'].tr("\n", "\t")}
#  </record>
#  remove_keys _hash, $.kubernetes, $.docker
#</filter>

<filter **>
  @type record_transformer
  auto_typecast false
  enable_ruby true
  <record>
    # use double quotes to avoid literal strings!
    log ${record['log'].tr("\n", "\t")}
  </record>
  remove_keys _hash, $.kubernetes, $.docker
</filter>
```

**Comment by hostalp** [comment]
I'd also like to see this possible with `whitelist_keys`

**Comment by dkulchinsky** [comment]
@repeatedly this would be an awesome enhancement for this plugin! 🙏🏼

**Comment by repeatedly** [comment]
Sorry for the delay. I have several private tasks after this PR and forgot to merge.
Just merged and released 2.2.0 with this patch :)

**Comment by dkulchinsky** [comment]
Thanks a lot @repeatedly!
